services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/ProjectSettingsManagerImpl.java

```diff
@@ -189,25 +195,33 @@ public void deleteProjectSetting(UserInfo userInfo, String id) throws DatastoreE
 			//A valid bucket name must also be a valid domain name
 			ValidateArgument.requirement(InternetDomainName.isValid(externalS3StorageLocationSetting.getBucket()), "Invalid Bucket Name");
 
-			validateBucketAccess(externalS3StorageLocationSetting);
-			validateOwnership(externalS3StorageLocationSetting, userProfile);
+			validateS3BucketAccess(externalS3StorageLocationSetting);
+			validateS3BucketOwnership(externalS3StorageLocationSetting, userProfile);
+		} else if (storageLocationSetting instanceof ExternalGoogleCloudStorageLocationSetting) {
+			UserProfile userProfile = userProfileManager.getUserProfile(userInfo.getId().toString());
+			ExternalGoogleCloudStorageLocationSetting externalGoogleCloudStorageLocationSetting = (ExternalGoogleCloudStorageLocationSetting) storageLocationSetting;
+			//A valid bucket name must also be a valid domain name
+			ValidateArgument.requirement(InternetDomainName.isValid(externalGoogleCloudStorageLocationSetting.getBucket()), "Invalid Bucket Name");
```

Comment:
maybe use the google regex for bucket names here.

Reply:
Not in their docs but in their Bucket creation wizard, Google requires bucket names are valid domain names, so this should be sufficient.